odoo/models.py

```diff
@@ -3531,13 +3531,22 @@ def get_metadata(self):
             res = self.read(LOG_ACCESS_COLUMNS)
         else:
             res = [{'id': x} for x in self.ids]
-        xml_data = dict((x['res_id'], x) for x in IrModelData.search_read([('model', '=', self._name),
-                                                                           ('res_id', 'in', self.ids)],
-                                                                          ['res_id', 'noupdate', 'module', 'name'],
-                                                                          order='id DESC'))
+
+        imds = IrModelData.search_read(
+            [('model', '=', self._name), ('res_id', 'in', self.ids)],
+            ['res_id', 'noupdate', 'module', 'name'],
+            order='id DESC'
+        )
+
+        xml_data = dict((x['res_id'], x) for x in imds)
```

Comment:
Might be a better idea to accumulate to a multidict (either "real" or just a `defaultdict(list)`) to avoid the extra expense when retrieving all the xids?

In that case the reference xid would be at `-1`. Or we reverse the ordering, and use the id `0` instead, either would work.

Reply:
I updated as suggested.

Keeping same default order for search, and reversed it later...
It avoids to change the behavior/order for record.get_metadata()[0]['xmlid']

Ty for review.